dashboard/src/components/SideMenu/SideMenu.tsx

```diff
@@ -65,6 +48,43 @@ const SideMenu = (): JSX.Element => {
   const notSelectedItemClassName =
     'w-full flex pl-5 py-4 cursor-pointer text-white';
 
+  const navigate = useNavigate();
+
+  const useNavigateTo = (path: string): (() => void) =>
+    useCallback(() => {
+      const queryParams = new URLSearchParams(location.search);
```

Comment:
tanstack router is type safe so please use it to get the search parameters

https://tanstack.com/router/latest/docs/framework/react/guide/search-params